src/DotNetLightning.Core/Utils/Primitives.fs

```diff
@@ -117,30 +120,43 @@ module Primitives =
             ec |> LNECDSASignature
 
 
-    type PaymentPreimage = | PaymentPreimage of byte [] with
-        member x.Value = let (PaymentPreimage v) = x in v
+    type PaymentHash = | PaymentHash of uint256 with
+        member x.Value = let (PaymentHash v) = x in v
+        member x.ToBytes(?lEndian) =
+            let e = defaultArg lEndian true
+            x.Value.ToBytes(e)
+
+        member x.GetRIPEMD160() =
+            let b = x.Value.ToBytes()
+            Crypto.Hashes.RIPEMD160(b, b.Length)
+
+
+    type PaymentPreimage(value: seq<byte>) =
```

Comment:
I prefer to use single case union. so that it aligns well with other types defined in this file.

Reply:
Out of curiosity, any special reason why a single case union would be better? Just for consistency or more reasons?

Reply:
I love automatic equality and comparison implementation in single case union.
serialization property test will be easy, for example.

Not sure that is the case for this one though.

Reply:
Ah good point!